hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenBurnHandler.java

```diff
@@ -36,10 +56,14 @@
  * HederaFunctionality#TOKEN_BURN}.
  */
 @Singleton
-public class TokenBurnHandler implements TransactionHandler {
+public class TokenBurnHandler extends BaseTokenHandler implements TransactionHandler {
```

Comment:
nit: Should this (and other handlers) be final classes?

Reply:
Good question. Probably. I don't see any reason why we would ever want to override them, except _maybe_ in testing, and even then it seems like we could just write a new class with the same interface

Reply:
Done

Reply:
Ironically I just found a case where it might be useful. Token wipe is very similar to token burn, except the wipe also reduces the token supply. Not sure if this is a good idea, but the wipe handler could be a child class of the burn handler and wraps a call to its parent, then does its wipe-specific handling. 

That said, kinda rubs me the wrong way for some reason. An alternative is to put the checks similar to burn and wipe into a different class that both handlers then call, and that fits better with 'prefer composition over inheritance.'  I dunno, might be overthinking it. Thoughts?